### ISSUE TITLE
Plate viewing in MoBIE

### ISSUE THREAD

**Issue by tischi** [mobie/mobie-viewer-fiji#977]
@tibuch 

How does the raw data look like that you want a plate view of?

Maybe we could directly open it without needing to convert to OME-Zarr:

https://mobie.github.io/tutorials/hcs.html

If you could share two wells of that data with me I could give it a shot!

**Comment by tibuch** [comment]
Hey @tischi,

currently we are working on Molecular Devices ImageXpress data. This data has to be converted into something reasonable to view anyway and hence converting it to ome-zarr is perfectly fine. It also works very nicely :) Currently working on a little package and example notebook here: https://github.com/fmi-faim/faim-hcs/blob/dev/examples/Create%20ome-zarr%20from%20Single-Plane%20Multi-Field%20Acquisition.ipynb

In the repo should also be some example data. Would this be enough for your test?

**Comment by tischi** [comment]
Is this the test data?

https://github.com/fmi-faim/faim-hcs/tree/dev/resources/MIP-2P-2sub/2022-07-05/1075

**Comment by tibuch** [comment]
Yes. This might help understand it: https://github.com/fmi-faim/faim-hcs/blob/95c6d3e05006aa9f2855e314925954096522caca/src/faim_hcs/io/MolecularDevicesImageXpress.py#L9

**Comment by tischi** [comment]
@tibuch

Is that how it should look like?

<img width="475" alt="image" src="https://user-images.githubusercontent.com/2157566/222164346-ffd880c1-2902-4647-9f2b-2981e7f7031a.png">

**Comment by tibuch** [comment]
Hmm, no the zarr looks like this:
![image](https://user-images.githubusercontent.com/5450869/222165343-27a040cc-9ef5-4eac-94dc-566a8636ff6e.png)
The fields (sub-positions) should be next to each other.

**Comment by tibuch** [comment]
This is how the image is montaged: https://github.com/fmi-faim/faim-hcs/blob/dev/src/faim_hcs/MetaSeriesUtils.py#L116

**Comment by tischi** [comment]
Hmm, something is still going wrong (or we don't use the same data):

<img width="1000" alt="image" src="https://user-images.githubusercontent.com/2157566/222186869-7fb64de3-32e1-44b4-b5ec-e6a96df1de71.png">

I am using this:

```
1075 (dev) $ ls
total 65M
drwxr-xr-x 18 tischer  576 Mar  1 14:59 ./
drwxr-xr-x  3 tischer   96 Mar  1 14:59 ../
-rwxr-xr-x  1 tischer 8.1M Mar  1 14:59 MIP-2P-2sub_C05_s1_w146C9B2CD-0BB3-4B8A-9187-2805F4C90506.tif*
-rwxr-xr-x  1 tischer  16K Mar  1 14:59 MIP-2P-2sub_C05_s1_w1_thumb6EFE77C6-B96D-412A-9FD1-710DBDA32821.tif*
-rwxr-xr-x  1 tischer 8.1M Mar  1 14:59 MIP-2P-2sub_C05_s1_w2B90625C8-6EA7-4E54-8289-C539EB75263E.tif*
-rwxr-xr-x  1 tischer  16K Mar  1 14:59 MIP-2P-2sub_C05_s1_w2_thumbEDDF803A-AE5E-4190-8C06-F54341AEC4A6.tif*
-rwxr-xr-x  1 tischer 8.1M Mar  1 14:59 MIP-2P-2sub_C05_s2_w1E2913F7F-E229-4B6A-BFED-02BCF54561FA.tif*
-rwxr-xr-x  1 tischer  16K Mar  1 14:59 MIP-2P-2sub_C05_s2_w1_thumb72E3641A-C91B-4501-900A-245BAC58FF46.tif*
-rwxr-xr-x  1 tischer 8.1M Mar  1 14:59 MIP-2P-2sub_C05_s2_w241C38630-BCFD-4393-8706-58755CECE059.tif*
-rwxr-xr-x  1 tischer  16K Mar  1 14:59 MIP-2P-2sub_C05_s2_w2_thumb5377A5AC-9BBF-4BAF-99A2-24896E3373A2.tif*
-rwxr-xr-x  1 tischer 8.1M Mar  1 14:59 MIP-2P-2sub_C06_s1_w152C23B9A-EB4C-4DF6-8A7F-F4147A9E7DDE.tif*
-rwxr-xr-x  1 tischer  16K Mar  1 14:59 MIP-2P-2sub_C06_s1_w1_thumb541AA634-387C-4B84-B0D8-EE4CB1C88E81.tif*
-rwxr-xr-x  1 tischer 8.1M Mar  1 14:59 MIP-2P-2sub_C06_s1_w2FB0D7D9B-3EA0-445E-9A05-7D01154A9A5C.tif*
-rwxr-xr-x  1 tischer  16K Mar  1 14:59 MIP-2P-2sub_C06_s1_w2_thumb8FA1E466-57CD-4237-B09B-CAB48154647D.tif*
-rwxr-xr-x  1 tischer 8.1M Mar  1 14:59 MIP-2P-2sub_C06_s2_w1F365E60C-BCC2-4B74-9856-BCE07C8B0FD3.tif*
-rwxr-xr-x  1 tischer  16K Mar  1 14:59 MIP-2P-2sub_C06_s2_w1_thumb9652366E-36A0-4B7F-8B18-DA89D7DB41BD.tif*
-rwxr-xr-x  1 tischer 8.1M Mar  1 14:59 MIP-2P-2sub_C06_s2_w20EEC6AEA-1727-41E6-806C-40FF6AF68B6C.tif*
-rwxr-xr-x  1 tischer  16K Mar  1 14:59 MIP-2P-2sub_C06_s2_w2_thumb710CD846-0185-4362-BBAF-C700AE0013B3.tif*
```

**Comment by tischi** [comment]
The two images on the far left and far right are the same as for you, but the middle ones looks wrong.

**Comment by tibuch** [comment]
Cool! You create now a view for each image? Would this also translate to 3D, where Molecular Devices saves each z-plane in a different sub-directory? What is your feeling performance wise?

**Comment by tischi** [comment]
It should be now on MoBIE-beta such that you can test yourself in Fiji.

`Plugins › MoBIE › Open › Open HCS Dataset...`

<img width="916" alt="image" src="https://user-images.githubusercontent.com/2157566/222189007-3a44bb56-1678-4ad7-b155-bec63d7baa4b.png">

For supporting also Z planes please open a separate issue with a link to example data.

Let me know how the performance is!

**Comment by tibuch** [comment]
Nice! It works, but not having a resolution pyramid is sub-optimal. I loaded a 384 well plate and my memory filled up completely and impacted the navigation speed. Not sure how the performance would be with enough memory to hold everything.

**Comment by tischi** [comment]
Yes, if you zoom out then I think it _cannot_ work because it needs to load everything.
But if you don't zoom out all the way (and use the table for jumping from well to well) I guess it is OK?

**Comment by tibuch** [comment]
Without zooming out it works. But still it is considerably slower than saving everything as ome-zarr (the four sub-positions are stitched and then saved as a single field in ome-zarr).

**Comment by constantinpape** [comment]
> But still it is considerably slower than saving everything as ome-zarr (the four sub-positions are stitched and then saved as a single field in ome-zarr).

Just to add, I don't think our aim here is to replace the full capabilities of resaving things as ome.zarr, but rather provide an additional option with less functionality (and worse performance for large data), that can however be used without changing the raw data at all.

**Comment by tischi** [comment]
Yes exactly, because re-saving the could mean duplicating it. At least currently, where, e.g. CellProfiler is not able to analyze an OME-Zarr plate, afaik...

**Comment by tibuch** [comment]
@constantinpape and @tischi these are two excellent points and this goal is certainly achieved! Thanks for developing MoBIE :1st_place_medal:

**Comment by constantinpape** [comment]
And we'll have the best of both worlds once microscopes write to ome-zarr directly ;)
https://github.com/mobie/mobie-viewer-fiji/issues/984
